src/reducers/functions/shape.js

```diff
@@ -0,0 +1,53 @@
+import guid from 'guid';
+
+export function resizeShape(state, action) {
+    // Create a copy of the drawing and then add the new shape:
+    const updatedState = Object.assign({}, state);
+
+    updatedState.drawing = Object.assign({}, state.drawing);
+
+    const shape = Object.assign({}, state.drawing[state.selected[0]]);
+
+    if (action.shape.moveDeltaX > 0) {
+        shape.width = action.shape.moveDeltaX;
+    } else {
```

Comment:
We can probably just remove the else statement. I can't imagine what we'd want to do otherwise